Add cross-support for Scala.js 1

lolgab opened the pull request:
Still requires shapeless to be published.

lolgab committed:
Add crosssupport for Scala.js 1.0.1

mikla commented:
btw: shapeless 2.3.3 is cross-published for Scala.js 1.

lolgab commented:
Updated the PR but it's still missing [spire](https://github.com/typelevel/spire)

mikla commented:
@lolgab spire `0.17.0-RC1` is published for sjs 1 🥳

lolgab committed:
Update spire to 0.17.0-RC1

- Update ScalaJS to 1.1.1

lolgab committed:
Update Scala versions

- Update .travis.yml and fix deprecations

lolgab committed:
Add SCALAJS_VERSION to .travis.yml

lolgab commented:
@mikla Perfect, now all tests pass with both Scala.js versions. I'm ready for review @rudogma .
Is it possible to run travis on forks?

lolgab committed:
Skipping tests in Scala 2.11.12 because spire dropped support

mikla commented on .travis.yml:
```diff
@@ -1,10 +1,17 @@
+    - $HOME/.cache/coursier
+    - $HOME/.sbt/boot
+script:
+  - sbt clean test
```
shouldn't it be `- sbt ++$TRAVIS_SCALA_VERSION clean test` ?

mikla replied:
Nevermind, your version also works.

lolgab replied:
Right! 🙂 
Fixed in the last commit.

mikla commented:
@lolgab you build is green https://travis-ci.org/github/rudogma/scala-supertagged
🥳 🚀

lolgab committed:
Add ++$TRAVIS_SCALA_VERSION to travis

lolgab committed:
Move tests to separate module

Setting Compile / crossScalaVersions doesn't work with the ++version syntax
Tests now run on Scala 2.12 and 2.13 while 2.11 is still supported on compile

lolgab committed:
Make supertagged a CrossType.Pure crossProject

- Move the crossScalaVersions settings to a separate object

lolgab committed:
Set Scala 2.11 version to 2.11.12

- Fix problems related with crossScalaVersions

mikla commented on project/Build.scala:
```diff
@@ -5,9 +5,9 @@ import sbt._
-  val Scala211 = "2.11.11"
-  val Scala212 = "2.12.10"
-  val Scala213 = "2.13.1"
+  val Scala211 = "2.11.12"
```
spire is not published for 2.11. 

https://mvnrepository.com/artifact/org.typelevel/spire

So, i guess it's time to drop 2.11 support.

lolgab replied:
Spire is used only in tests. I tweaked the build to skip tests for Scala 2.11 in case the author wants to maintain Scala 2.11 / Scala Native.

fdietze commented:
Great work everyone!

Until this PR is merged, you can use it already as a jitpack dependency:

```scala
resolvers += "jitpack" at "https://jitpack.io"
libraryDependencies += "com.github.lolgab" % "scala-supertagged" % "scalajs-1.0.0-RC2-SNAPSHOT"
```

mikla commented:
@fdietze thanks.
Maybe you can also publish it for Scala 2.13?

mikla commented:
@rudogma could you please have a look this MR?

fdietze commented:
@mikla I updated the dependency code.

rudogma commented:
> @rudogma could you please have a look this MR?

@mikla , @lolgab thx for pull. Will take a look in a few days and publish it.
